velox/vector/BaseVector.cpp

```diff
@@ -1185,9 +1185,10 @@ uint64_t BaseVector::estimateFlatSize() const {
   // size times this vector's size plus any nulls of this vector.
   if (UNLIKELY(leaf->size() == 0)) {
     const auto& leafType = leaf->type();
+    uint64_t stringBufferSize{0};
     return length_ *
         (leafType->isFixedWidth() ? leafType->cppSizeInBytes() : 0) +
-        BaseVector::retainedSize();
+        BaseVector::retainedSizeImpl(stringBufferSize);
```

Comment:
Perhaps, BaseVector::retainedSizeImpl() can be a non-virtual method with no arguments. Then, there will be a pure virtual method with a single argument.